hardware_interface/source/eps.c

```diff
@@ -74,6 +74,77 @@ eps_instantaneous_telemetry_t get_eps_instantaneous_telemetry() {
     return telembuf;
 }
 
+/**
+ * @brief
+ *      Electronic Power System get Housekeeping data
+ * @details
+ *      May replace get_eps_instantaneous_telemetry() if deemed a suitable 
+ *      replacement
+ * @param telembuf
+ *      pointer to struct
+ * 
+ * @return 
+ *      Will likely want to return at least a boolean for success
+ *      but currently no return 
+ * 
+ */
+void EPS_getHK(eps_instantaneous_telemetry_t* telembuf) {
+    eps_t *eps;
+    eps = prv_get_eps();
+    prv_get_lock(eps);
+    configASSERT(eps);
+
+    telembuf->cmd = eps->hk_telemetery.cmd;
+    telembuf->status = eps->hk_telemetery.status;
+    telembuf->vBatt = eps->hk_telemetery.vBatt;
+    telembuf->curSolar = eps->hk_telemetery.curSolar;
+    telembuf->curBattIn = eps->hk_telemetery.curBattIn;
+    telembuf->curBattOut = eps->hk_telemetery.curBattOut;
+    telembuf->reserved1 = eps->hk_telemetery.reserved1;
+    telembuf->outputStatus = eps->hk_telemetery.outputStatus;
+    telembuf->outputFaultStatus = eps->hk_telemetery.outputFaultStatus;
+    telembuf->wdt_gs_time_left = eps->hk_telemetery.wdt_gs_time_left;
+    telembuf->wdt_gs_counter = eps->hk_telemetery.wdt_gs_counter;
+    telembuf->rstReason = eps->hk_telemetery.rstReason;
+    telembuf->bootCnt = eps->hk_telemetery.bootCnt;
+    telembuf->battMode = eps->hk_telemetery.battMode;
+    telembuf->mpptMode = eps->hk_telemetery.mpptMode;
+    telembuf->batHeaterMode = eps->hk_telemetery.batHeaterMode;
+    telembuf->batHeaterState = eps->hk_telemetery.batHeaterState;
+    telembuf->reserved5 = eps->hk_telemetery.reserved5;

```

Comment:
Why not memcpy?

Reply:
Memcpy just copies all the data addresses and leaves a shallow copy. This risks the data being changed unexpectedly or deallocated before it can be used. To avoid that, a deep copy is created in this manner.

Reply:
Well, the bad news is that these are subject to change, based on what I'm receiving from EPS right now. So, we'll have to rewrite some stuff here. But we can worry about that later.

Reply:
Yes. Most of the data from all the subservices are. Thankfully changes can be isolated to a few spots. There's just no generic way to deep copy complex structs so we're stuck with it unless we risk shallow copies. Which we could do. But it brings in other risks such as unexpected NULL pointers